packages/wonder-blocks-form/src/components/labeled-text-field.js

```diff
@@ -41,6 +41,27 @@ type Props = {|
      */
     disabled: boolean,
 
+    /**
+     * Whether this field is required to to continue.
+     *
+     * If the `validate` prop is not being passed in in conjuction with
+     * the `required` prop, please use the `requiredErrorMessage` prop
+     * to specify the error message.
+     */
+    required?: boolean,
+
+    /**
+     * The error message that should render if this field is
+     * required and left blank. Note that this message will not be used
+     * if a `validate` prop is passed in.
+     *
+     * Implementation note: Added this field so that users can pass in
+     * translated strings.
+     *
+     * Example message: i18n._("A password is required to log in.")
+     */
+    requiredErrorMessage?: string,
```

Comment:
Should the `required` prop be merged with this one since this error message is meaningless without the field being required?

Perhaps:

```
type Required = boolean | string;

type Props = {|
    ....
    required?: Required,
    ....
```

That way it can either be true/falsy or a string that is the error.